kof-operator/internal/controller/clusterdeployment_controller_test.go

```diff
@@ -460,5 +461,57 @@ var _ = Describe("ClusterDeployment Controller", func() {
 			Expect(err).NotTo(HaveOccurred())
 			Expect(configMapCDGeneration).To(BeNumerically(">", initialClusterDeploymentGeneration))
 		})
+
+		It("should discover regional cluster by AWS region", func() {
+			By("creating child ClusterDeployment without kof-regional-cluster-name label")
+			const childClusterDeploymentName = "test-child-aws"
+
+			const childClusterDeploymentConfig = `{
+				"region": "us-east-2",
+				"clusterLabels": {"k0rdent.mirantis.com/kof-cluster-role": "child"}
+			}`
+
+			childClusterConfigMapNamespacedName := types.NamespacedName{
+				Name:      "kof-cluster-config-" + childClusterDeploymentName,
+				Namespace: DEFAULT_NAMESPACE,
+			}
+
+			childClusterDeploymentNamespacedName := types.NamespacedName{
+				Name:      childClusterDeploymentName,
+				Namespace: DEFAULT_NAMESPACE,
+			}
+
+			createClusterDeployment(
+				childClusterDeploymentName,
+				childClusterDeploymentConfig,
+			)
+
+			DeferCleanup(func() {
```

Comment:
We do cleanup of childClusterDeployment in AfterEach block.
Why do we need a cleanup here?